zio-http/src/main/scala/zio/http/netty/NettyResponseEncoder.scala

```diff
@@ -0,0 +1,56 @@
+package zio.http.netty
+
+import io.netty.buffer.Unpooled
+import io.netty.handler.codec.http._
+import zio._
+import zio.http._
+
+import java.util.concurrent.ConcurrentHashMap
+
+private[zio] object NettyResponseEncoder {
+
+  private val frozenCache = new ConcurrentHashMap[Response, HttpResponse]()
```

Comment:
Consider:
```suggestion
  private val frozenCache = new ConcurrentHashMap[Response, UIO[HttpResponse]]()
```

This way you can put the `HttpResponse` into `ZIO.succeedNow` and return it directly from `encode`.

Reply:
There are places where `fastEncode` is used where it completely bypasses needing ZIO runtime.  Wrapping the cached response in `UIO` would prevent us from using these optimized code pathes.

Alternatively, we could store a tuple `(HttpResponse, UIO[HttpResponse])` that way we get the best of both approach with an additional allocation of a tuple.